### PR TITLE
dts: Rename generated_dts_board* to devicetree*, with backwards compatibility

### PR DESCRIPTION
**Pull request by ulfalizer**
Three commits:

```
generated_dts_board.h is pretty redundant and confusing as a name. Call
it devicetree.h instead.

dts.h would be another option, but DTS stands for "devicetree source"
and is the source code format, so it's a bit confusing too.

The replacement was done by grepping for 'generated_dts_board' and
'GENERATED_DTS_BOARD'.

Two build diagram and input-output SVG files were updated as well, along
with misc. documentation.

hal_ti, mcuboot, and ci-tools updates are included too, in the west.yml
update.
```

```
dts: Add include/generated_dts_board.h for backwards compatibility

Includes include/devicetree.h and generates a warning.
```

```
doc: release-notes-2.2: Mention generated_dts_board* renaming

generated_dts_board*.{h,conf} have been renamed to devicetree*.{h,conf}.
```

### CODE REVIEW

**Review by cvinayak — APPROVED** [review]
QUICK ACK.

**Comment by ulfalizer** [comment]
Needs https://github.com/zephyrproject-rtos/ci-tools/pull/111 and https://github.com/zephyrproject-rtos/hal_ti/pull/3. Multi-repo makes changes like these awkward.

**Comment by aescolar** [comment]
@erwango, @rlubos, @dcpleung, @nategraff-sifive and @wentongwu : Sorry, the list of reviewers was over the maximum allowed and trying to edit it caused it to be truncated

**Comment by George-Stefan** [comment]
@alexandru-porosanu-nxp ^^

**Review by galak — DISMISSED** [review]
@SebastianBoe had some desire that we use toolchain_cc_imacros() to get devicetree.h included the same way we include `${AUTOCONF_H}` so we don't need to do the `#include`.

So lets close on that before we rename everything vs removing those lines.

**Comment by ulfalizer** [comment]
> @SebastianBoe had some desire that we use toolchain_cc_imacros() to get devicetree.h included the same way we include `${AUTOCONF_H}` so we don't need to do the `#include`.
> 
> So lets close on that before we rename everything vs removing those lines.

Renaming is still worthwhile to reduce developer confusion I think, but no panic at least.

**Comment by galak** [comment]
> Renaming is still worthwhile to reduce developer confusion I think, but no panic at least.

Yeah, I'm not opposed to the file renaming.  Just figure if we're going to touch all the files that include things likes decide how we want to really do it, and make that part of the change once.  see issue #7214

**Review by dbkinder — APPROVED** [review]
doc changes LGTM (and isn't' it nice to be able to grep and edit SVG images :)

**Comment by galak** [comment]
Can you rebase.

File: include/devicetree.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) 2019 Nordic Semiconductor
+ *
+ * Not a generated file. Feel free to modify.
+ */
+
+#ifndef DEVICETREE_H
+#define DEVICETREE_H
+
+#include <devicetree_unfixed.h>
+#include <devicetree_unfixed.h.deprecated>
```
Comment (SebastianBoe) [inline include/devicetree.h]: This is no longer present is it?

Comment (ulfalizer) [reply include/devicetree.h]: Nope. Fixed.

**Comment by ulfalizer** [comment]
> Can you rebase.

Rebased now.

**Comment by ulfalizer** [comment]
@galak 
ping

**Comment by carlescufi** [comment]
> @SebastianBoe had some desire that we use toolchain_cc_imacros() to get devicetree.h included the same way we include `${AUTOCONF_H}` so we don't need to do the `#include`.
> 
> So lets close on that before we rename everything vs removing those lines.

Is this still the case @SebastianBoe?

**Comment by ulfalizer** [comment]
The way to merge this is to first merge https://github.com/zephyrproject-rtos/ci-tools/pull/111. I'll then update this PR to also add the ci-tools change.

`GENERATED_DTS_BOARD` needs to be renamed to `DEVICETREE_CONF` simultaneously in both repos.

**Comment by SebastianBoe** [comment]
> @SebastianBoe had some desire that we use toolchain_cc_imacros() to get devicetree.h included the same way we include ${AUTOCONF_H} so we don't need to do the #include.

This was based on a model where the user could consider DT and Kconfig symbols to be originating from two different sources based on the prefix, but otherwise be two equal configuration systems that are merged together and available in a consistent way.

But we have now moved DT out of CMake, so these symbols are not available simultaneously and it is not a reasonable assumption that they are available through the same mechanisms.

Including devicetree.h through the preprocessor is reasonable since using "-include" is not without issues (it causes trouble for static analyzers, IDE's, programmers looking for where a symbol comes frlom, and other tooling).

**Comment by ulfalizer** [comment]
Rebased again. Some headers got moved around.

@carlescufi 
See https://github.com/zephyrproject-rtos/zephyr/pull/20757#issuecomment-567261796 for how to merge.

**Review by carlescufi — APPROVED** [review]
I am OK with this change as it is now after @SebastianBoe 's explanation.

@nashif are you ok with the procedure described [here](https://github.com/zephyrproject-rtos/zephyr/pull/20757#issuecomment-567261796)?

EDIT; @ulfalizer you will need an update to `west.yml` to reflect the changes in hal_ti

**Comment by tbursztyka** [comment]
sounds like a relevant change to me as well. +1

**Comment by ulfalizer** [comment]
@carlescufi 
> EDIT; @ulfalizer you will need an update to `west.yml` to reflect the changes in hal_ti

Can't update `west.yml` until the commits are in. See https://github.com/zephyrproject-rtos/zephyr/pull/20757#issuecomment-567261796 for instructions.

**Comment by ulfalizer** [comment]
Updated instructions (ci-tools change got in):

1. Merge https://github.com/zephyrproject-rtos/hal_ti/pull/3
2. Merge https://github.com/zephyrproject-rtos/mcuboot/pull/2
3. I'll update west.yml here
4. Merge this

File: west.yml [hunk 2]
```diff
@@ -74,7 +74,7 @@ manifest:
       revision: 3776c158fe138a72c97c187e4d31c81c37884be9
       path: modules/crypto/mbedtls
     - name: mcuboot
-      revision: c2bd757332bfde11c27019f937b0c61d5e9b9fd9
+      revision: pull/2/head
```
Comment (nvlsianpu) [inline west.yml]: I synchronized zephyr-fork mcuboot, use SHA : e96b6873bd8a64185796ac47b941f56dd29b6e3f

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: west.yml (hunk 2 above)
Comment (carlescufi) [inline west.yml]: Please replace with e96b6873bd8a64185796ac47b941f56dd29b6e3f, now that the required fix is merged

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by ulfalizer** [comment]
Rebased now. Had a bunch of clashes in docs and elsewhere.

Also updated the diagrams with https://draw.io. For whatever reason, some of the SVG files are included directly in the documentation (`doc/guides/build/build-config-phase.svg`), while others have been converted to PNG files that are included instead (the ones in `doc/guides/dts/`).

A messy thing about `doc/guides/dts/zephyr_dt_inputs_outputs.png` is that it's not synced to `doc/guides/dts/zephyr_dt_inputs_outputs.svg`. That's why the text is a bit different in the updated version.

These diagrams have some errors and confusing stuff in them. I'll clean them separately later. I mostly just renamed `generated_dts*` to `devicetree*` here.

`doc/guides/build/build-config-phase.svg` before/after:

config-before

config-after

`doc/guides/dts/zephyr_dt_inputs_outputs.svg` before/after:

devicetree-before

devicetree-after

**Review by nvlsianpu — APPROVED** [review]

**Comment by mbolivar** [comment]
Very nice work!

**Comment by ulfalizer** [comment]
Needs https://github.com/zephyrproject-rtos/ci-tools/pull/117 too.

**Comment by ulfalizer** [comment]
> Needs [zephyrproject-rtos/ci-tools#117](https://github.com/zephyrproject-rtos/ci-tools/pull/117) too.

In now.

**Pull request merged**